core/src/main/java/com/google/instrumentation/trace/TraceExporter.java

```diff
@@ -54,6 +71,352 @@
+           * @throws IllegalArgumentException if {@code numSamples} is negative.
+           */
+          public static ErrorBucketSummary create(int numSamples, CanonicalCode canonicalCode) {
+            checkArgument(numSamples >= 0, "Negative numSamples.");
+            //CHECKSTYLE:OFF: Long class name.
+            return new AutoValue_TraceExporter_InProcessDebuggingHandler_Summary_PerSpanNameSummary_ErrorBucketSummary(
+                numSamples, canonicalCode);
+            //CHECKSTYLE:ON: Long class name.
+          }
+
+          /**
+           * Returns the number of sampled spans in this bucket.
+           *
+           * @return the number of sampled spans in this bucket.
+           */
+          public abstract int getNumSamples();
+
+          /**
+           * Returns the {@code CanonicalCode} for this bucket. Always different than {@link
+           * CanonicalCode#OK}.
+           *
+           * @return the {@code CanonicalCode} for this bucket.
+           */
+          public abstract CanonicalCode getCanonicalCode();
+        }
+      }
+    }
+
+    /**
+     * Filter for active spans. Used to filter results returned by the {@link
+     * #getActiveSpans(ActiveSpansFilter)} request.
+     */
+    @AutoValue
+    @Immutable
+    public abstract static class ActiveSpansFilter {
+
+      ActiveSpansFilter() {}
+
+      /**
+       * Returns a new instance of {@code ActiveSpansFilter}.
+       *
+       * @param spanName the name of the span.
+       * @param maxSpansToReturn the maximum number of results to be returned. {@code 0} means all.
+       * @return a new instance of {@code ActiveSpansFilter}.
+       * @throws NullPointerException if {@code spanName} is {@code null}.
+       * @throws IllegalArgumentException if {@code maxSpansToReturn} is negative.
+       */
+      public static ActiveSpansFilter create(String spanName, int maxSpansToReturn) {
+        checkArgument(maxSpansToReturn >= 0, "Negative maxSpansToReturn.");
+        return new AutoValue_TraceExporter_InProcessDebuggingHandler_ActiveSpansFilter(
+            spanName, maxSpansToReturn);
+      }
+
+      /**
+       * Returns the span name.
+       *
+       * @return the span name.
+       */
+      public abstract String getSpanName();
+
+      /**
+       * Returns the maximum number of spans to be returned.
+       *
+       * @return the maximum number of spans to be returned.
+       */
+      public abstract int getMaxSpansToReturn();
+    }
+
+    /**
+     * Filter for latency based sampled spans. Used to filter results returned by the {@link
+     * #getLatencyBasedSampledSpans(LatencyBasedSampledSpansFilter)} request.
+     */
+    @AutoValue
+    @Immutable
+    public abstract static class LatencyBasedSampledSpansFilter {
+
+      LatencyBasedSampledSpansFilter() {}
+
+      /**
+       * Returns a new instance of {@code LatencyBasedSampledSpansFilter}.
+       *
+       * @param spanName the name of the span.
```

Comment:
I can't see any comment that says that filtering is based on the name. Please add a comment saying that, rather than leaving it implicit.